mantidproject/mantid
# ISIS Powder disable normalisation and empty corrections when current is 0

Previously, when processing runs with total current of 0, `isis_powder` scripts would crash with a divide by 0 error from `NormaliseByCurrent`. This is annoying if scientists are focusing and summing up a long list of runs, some of which may have no current - we just want these runs to contribute nothing to the final result.

When current is 0 (`tot_prtn_chrg` in sample logs), we don't bother running `NormaliseByCurrent`, and we also don't subtract the empty, as this gives us negative counts.

The end result should be:
- For `input_mode="summed"`, if the runs all have 0 current, the end result should be a workspace with 0 counts. If some of the runs have counts, the output workspace should be the same as if we hadn't included the empty runs
- For `input_mode="individual"`, we should be able to focus runs with 0 current on their own, and just get a workspace with 0 counts

**To test:**

If you're not at ISIS, you'll need to change the second line - ping me a message about setting up ISIS Powder directories, as it can be fiddly. Bear in mind `103106 has current, `103107` does not.
```
from isis_powder import Polaris
pol = Polaris(config_file="D:\ISIS_Powder\POLARIS\polaris_config.yaml",
              do_van_normalisation=False)
```
Focus a single workspace with 0 current, this not crash, and the output workspaces should be flat lines at 0
```
pol.focus(run_number="103107")
```
Now sum up `103107` and `1031081` - again, we should get workspaces with no counts
```
pol.focus(run_number="103107-103108")
```
Now focus `103106` and compare it to the result of summing `103106` with `103107` - they should be the same
```
individual = pol.focus(run_number="103106")
summed = pol.focus(run_number="103106-103107", input_mode="summed")
CompareWorkspaces(individual, summed)
```

Fixes #21805 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/compare/21805_isispowder_normalisebycurrent?expand=1#diff-6d316275f01d64b8113c082a773943ae)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
